specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/workspaceRP.json

```diff
@@ -4559,6 +4654,181 @@
         }
       }
     },
+    "EndpointModelDeprecationProperties": {
+      "type": "object",
+      "properties": {
+        "fineTune": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "fineTuneOn": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "inference": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "inferenceOn": {
+          "format": "date-time",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointModelProperties": {
+      "type": "object",
+      "properties": {
+        "capabilities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "deprecation": {
+          "$ref": "#/definitions/EndpointModelDeprecationProperties"
+        },
+        "finetuneCapabilities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "format": {
+          "type": "string"
+        },
+        "isDefaultVersion": {
+          "type": "boolean"
+        },
+        "lifecycleStatus": {
+          "type": "string"
+        },
+        "maxCapacity": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "name": {
+          "type": "string"
+        },
+        "skus": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointModelSkuProperties"
+          }
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/systemData"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    },
+    "EndpointModelSkuCapacityProperties": {
+      "type": "object",
+      "properties": {
+        "default": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "maximum": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "EndpointModelSkuProperties": {
+      "type": "object",
+      "properties": {
+        "capacity": {
+          "$ref": "#/definitions/EndpointModelSkuCapacityProperties"
+        },
+        "connectionIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "deprecationDate": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "deprecationOn": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "name": {
```

Comment:
>name [](http://example.com/codeflow?start=9&length=4)

You should rename this as it could be in conflict with top level properties with the same name

Reply:
https://github.com/Azure/azure-rest-api-specs/blob/2e5be0e72597c6fc8d438f20e38087d900c16427/specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2024-06-01-preview/cognitiveservices.json#L5458

trying to align schema from here